flink-tests/src/test/java/org/apache/flink/test/state/SavepointStateBackendSwitchTest.java

```diff
@@ -0,0 +1,289 @@
+        final KeyedStateHandle stateHandle = stateHandles.getJobManagerOwnedSnapshot();
+        try (final HeapKeyedStateBackend<String> keyedBackend =
+                createHeapKeyedStateBackend(StateObjectCollection.singleton(stateHandle))) {
+
+            InternalMapState<String, Integer, Long, Long> state =
+                    keyedBackend.createInternalState(IntSerializer.INSTANCE, mapStateDescriptor);
+
+            InternalValueState<String, Integer, Long> valueState =
+                    keyedBackend.createInternalState(IntSerializer.INSTANCE, valueStateDescriptor);
+
+            InternalListState<String, Integer, Long> listState =
+                    keyedBackend.createInternalState(IntSerializer.INSTANCE, listStateDescriptor);
+
+            keyedBackend.setCurrentKey("abc");
+            state.setCurrentNamespace(namespace1);
+            assertEquals(33L, (long) state.get(33L));
+            assertEquals(55L, (long) state.get(55L));
+            assertEquals(2, getStateSize(state));
+
+            state.setCurrentNamespace(namespace2);
+            assertEquals(22L, (long) state.get(22L));
+            assertEquals(11L, (long) state.get(11L));
+            assertEquals(2, getStateSize(state));
+            listState.setCurrentNamespace(namespace2);
+            assertThat(listState.get(), contains(4L, 5L, 6L));
+
+            state.setCurrentNamespace(namespace3);
+            assertEquals(44L, (long) state.get(44L));
+            assertEquals(1, getStateSize(state));
+
+            keyedBackend.setCurrentKey("mno");
+            state.setCurrentNamespace(namespace3);
+            assertEquals(11L, (long) state.get(11L));
+            assertEquals(22L, (long) state.get(22L));
+            assertEquals(33L, (long) state.get(33L));
+            assertEquals(44L, (long) state.get(44L));
+            assertEquals(55L, (long) state.get(55L));
+            assertEquals(5, getStateSize(state));
+            valueState.setCurrentNamespace(namespace3);
+            assertEquals(1239L, (long) valueState.value());
+            listState.setCurrentNamespace(namespace3);
+            assertThat(listState.get(), contains(1L, 2L, 3L));
+
+            KeyGroupedInternalPriorityQueue<TimerHeapInternalTimer<String, Integer>> priorityQueue =
+                    keyedBackend.create(
+                            "event-time",
+                            new TimerSerializer<>(
+                                    keyedBackend.getKeySerializer(), IntSerializer.INSTANCE));
+
+            assertThat(priorityQueue.size(), equalTo(1));
+            assertThat(
+                    priorityQueue.poll(),
+                    equalTo(new TimerHeapInternalTimer<>(1234L, "mno", namespace3)));
+        }
+    }
+
+    private HeapKeyedStateBackend<String> createHeapKeyedStateBackend(
+            Collection<KeyedStateHandle> stateHandles) throws BackendBuildingException {
+        // must be synchronized with the KeyGroup of the RocksDB state backend
```

Comment:
Should this be a constant in the test that is passed also when creating the Rocks backend?